### PR TITLE
Added DreamHost DNS API 

### PR DESCRIPTION
**Pull request by wickerwaka**
This is not original work, it is just a continuation of https://github.com/Neilpang/acme.sh/pull/902, whose work seems to be abandoned. I fixed up the CI errors and resolved some merge conflicts, nothing else.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -302,6 +302,7 @@ You don't have to do anything manually!
 1. Dynu API (https://www.dynu.com)
 1. DNSimple API
 1. NS1.com API
+1. DreamHost.com API
```
Comment (Neilpang) [inline README.md]: move  dreamhost to the last of the list.

File: dnsapi/dns_dreamhost.sh [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+  _debug url "$url"
+
+  if ! response="$(_get "$url")"; then
+    _err "Error <$1>"
+    return 1
+  fi
+
+  if [ -z "$2" ]; then
+    message="$(printf "%s" "$response" | _egrep_o "\"Message\":\"[^\"]*\"" | cut -d : -f 2 | tr -d \")"
```
Comment (Neilpang) [inline dnsapi/dns_dreamhost.sh]: don't  use `printf "%s"`,  just use `echo` is ok

**Comment by wickerwaka** [comment]
Not sure what the CI failure is, there are no logs.

**Comment by Neilpang** [comment]
@wickerwaka  it's not your fault.  please merge the latest dev code,

**Pull request merged**